inference-engine/src/cldnn_engine/cldnn_program.cpp

```diff
@@ -5271,6 +5274,35 @@ void Program::CreateCumSumPrimitive(cldnn::topology& topology, InferenceEngine::
     AddPrimitiveToProfiler(layerName, layer);
 }
 
+void Program::CreateRoundPrimitive(cldnn::topology& topology, InferenceEngine::CNNLayerPtr& layer) {
+    ValidateLayer(layer, 1);
+    auto inputPrimitives = GetPrevLayersPrimitives(layer);
+    auto layerName = layer_type_name_ID(layer);
+
+    std::string mode = layer->GetParamAsString("mode", "half_to_even");
+
+    if ((mode != "half_to_even") && (mode != "half_away_from_zero")) {
+        THROW_CLDNN_EXCEPTION("Unsupported mode (" + mode +
+                                  ") in layer " + layerName);
+    }
+    if (mode == "half_to_even") {
+        auto primitive = cldnn::activation(
+            layerName,
+            inputPrimitives[0],
+            cldnn::activation_func::round_half_to_even);
```

Comment:
primitive creation logic is duplicated. Could you convert std::string mode to corresponding activation_func and then create activation primitive once?